### ISSUE TITLE
0.4.1 Release

### ISSUE THREAD

**Issue by rhutchison** [meanjs/mean#833]
We are tracking towards the release of 0.4.1.  This thread is to discuss what needs to be finalized before we can cut the release.  If there are any PRs or open issues that should be added to the list, please comment below.

### General
 - [x] #759 [Enhancement] client-side validation @ilanbiala 
 - [x] #796 [Enhancement] http interceptor to handle unauthorized requests @ilanbiala 
 - [x] #843 [Bug] previous state history @lirantal
 - [x] #811 [Enhancement] code coverage report @lirantal 
 - [x] #836(#813/#814) XHTML compliant e-mail templates @lirantal 
 - [x] #830 [Bug] 6 character pw stored as plain text @lirantal
 - [x] #853 [Enhancement] Template cache @ilanbiala 
 - [x] #774 Remove data attribute prefix @ilanbiala 
 - [x] #815 CONTRIBUTING.md @lirantal
 - [x] #842 [Bug/Enhancement] Facebook Authentication @lirantal 
 - [x] #850 naming convention @lirantal 
 - [x] #868 Add Deploy to Bluemix button to documentation
 - [ ] ~~#636 Best Practices - Style Guide @ilanbiala~~
 
### Articles
 - [ ] ~~#800 route refactoring @ilanbiala~~
 - [ ] ~~#807 [Enhancement] admin @ilanbiala~~
   - ~~rebase after #800 is merged~~

**Comment by ilanbiala** [comment]
I don't think the style guide is making it into 0.4.1. Other than that, I'm fine with the rest of the list.

**Comment by simison** [comment]
Nice! What do you think about including #639?

**Comment by simison** [comment]
@ilanbiala do you recon style guide could do pre 0.5.0? I could start helping with it within a week or so (holidays over).

**Comment by rhutchison** [comment]
@ilanbiala other than it requires work, what is the reasoning for not setting styleguide as a stretch goal?

**Comment by mleanos** [comment]
@rhutchison I think the order which the Articles PR's should be merged is important #831 

I think development will go a lot fast once we get the Styling guide implemented. We'd end up having to re-do work if we waited.. If there's good reason to wait, then I say push it till after 0.5.0 or as a requirement for 0.5.0.

**Comment by mleanos** [comment]
@rhutchison I'm fine with #800 going in before #807 since it requires a bit more discussion.

**Comment by lirantal** [comment]
Thanks guys for summing this up.
Can you please add the assigned person (me/Ilan) per issue so we can easily navigate to relevant items? 

@ilanbiala you're welcome to add/remove items.
@rhutchison I'm overall in agreement with @ilanbiala about not pushing too big items such as the UX work. As you can see we're already doing *much* better in terms of agile and fast releases so it will be sure to catch up in 0.4.2 or something. The other reason would be to also let the changes quick in the dev cycle, meaning I would want to see UX story merged soon after 0.4.1 is out so we can "test drive" it for a little while on master before 0.4.2 is out.

**Comment by rhutchison** [comment]
@lirantal I never had the UX story in this list, I would like to see effort towards ngBestPractices though.  The UX can come in 0.4.3.

BTW, good work keeping the project moving - much improved

Also, I don't think non-contributors can manage/assign issues.

**Comment by lirantal** [comment]
Thanks, we're all doing a great job here, definitely feels like we should meet for some beer somehow ;)

With regards to the issues assigned, I meant to update them in the bullets list above so it's easier to see whose assigned to one in that list you've put up there.

**Comment by ilanbiala** [comment]
Sorry guys, but I have orientation week until Friday, so I'll be slow to get to these issues, though I will try my best. @lirantal if you want to take the lead on anything, I'm fine with that.

**Comment by lirantal** [comment]
OK then. I'll sync with the guys on each issue to move it further.

**Comment by ilanbiala** [comment]
Rearranged some issues so the 0.4.1 milestone contains all of the above issues/PRs. Let's try to release this in the next week.

**Comment by rhutchison** [comment]
Other than #811 is there anything else left open? @lirantal what is the status on #811?

I would recommend we cut 0.4.1 and move the articles PRs to 0.4.2.

Also starting with 0.4.2, all PRs with angular changes should follow angular-styleguide - if a file is touched, it should be updated.  I will put a larger effort to converting the modules.

**Comment by lirantal** [comment]
Hey guys,
Just merged #811.

@ilanbiala the rest are yours, are we going to merge them for 0.4.1?

**Comment by rhutchison** [comment]
@ilanbiala @lirantal please ship this as 0.4.1 and create the proper release tag.  We will start working on 0.4.2

**Comment by codydaig** [comment]
:+1:

**Comment by ilanbiala** [comment]
@lirantal yes 0.4.1 is good to go. The rest are going into 0.4.2.

**Comment by mleanos** [comment]
:+1: Nice. Great job everyone!

**Comment by ilanbiala** [comment]
@lirantal do you want to create the release and I'll work on the next PRs?

**Comment by ilanbiala** [comment]
Added #868 but it is already merged, so it should be good to go.

**Comment by lirantal** [comment]
@ilanbiala sure thing.

**Comment by lirantal** [comment]
@ilanbiala so basically except from tagging `master` branch as 0.4.1 there isn't anything else to do, right? we don't have an 0.4.1 branch like we had before with 0.4.0 so I don't see any merges required.

Tag created: https://github.com/meanjs/mean/releases/tag/v0.4.1

**Comment by ilanbiala** [comment]
Yeah I think we are good. @rhutchison what's the integration branch for?

**Comment by lirantal** [comment]
Merged related PRs https://github.com/meanjs/mean/pull/877 and https://github.com/meanjs/mean/pull/876 to update the version number to 0.4.1

**Comment by rhutchison** [comment]
@ilanbiala it was to help test a bunch of related PRs.  It's removed now.

**Comment by rhutchison** [comment]
Thanks @lirantal.  Please recreate the tag, now that #876 is merged and close this issue when you are done.

We should introduce CHANGELOG.md in next release.

**Comment by lirantal** [comment]
Done.

**Comment by codydaig** [comment]
@rhutchison Chanelog.html is just in the gh_pages branch.

**Comment by ilanbiala** [comment]
@rhutchison I'd like to start the CHANGELOG.md from 0.4 as the "initial release" so we don't have to compare it against 0.3.

**Comment by lirantal** [comment]
I think we can close this can move the discussion to https://github.com/meanjs/mean/issues/881 and the 0.4.2 issue
